src/main/java/com/newts/newtapp/api/application/conversation/RemoveUser.java

```diff
@@ -40,11 +40,17 @@ public Void request(RequestModel request) throws UserNotFound, ConversationNotFo
 
         // Check that the user is in the conversation
         if (conversation.getUsers().contains(userID)) {
-            // Remove the user from the conversation
-            conversation.removeUser(user);
-            user.removeConversation(conversation);
-            conversationRepository.save(conversation);
-            userRepository.save(user);
+            // Checks to see how many users are in conversation, if 1 delete conversation
+            // else ust remove user from conversation.
+            if(conversation.getNumUsers() == 1){
+                conversationRepository.delete(conversation);
+                conversationRepository.save(conversation);
```

Comment:
If saving the conversation will add it to the database again after you just deleted it. you don't need line 47

Reply:
Good to merge after this is resolved!